tft.py

```diff
@@ -83,32 +83,34 @@ def league_client_running() -> bool:
     ) and system_helpers.find_in_processes(CONSTANTS["executables"]["league"]["client_ux"])
 
 
-def parse_task_kill_text(result: subprocess.CompletedProcess[str]) -> None:
+def parse_task_kill_result(result: subprocess.CompletedProcess[str]) -> None:
     """Parses the task kill text to log the result.
 
     Args:
         result (subprocess.CompletedProcess[str]): the return value of the subprocess run.
     """
-    if "not found." in result.stderr:
-        logger.debug(f"{result.args[-1]} was not running.")
-    elif "has been terminated." in result.stdout:
-        logger.debug(f"{result.args[-1]} has been terminated.")
+    process_name = result.args.replace("taskkill ", "").replace("/f ", "").replace("/im ", "")
+    if result.returncode == 128:
+        logger.debug(f"{process_name} was not running.")
+    elif result.returncode == 0:
+        logger.debug(f"{process_name} has been terminated.")
     else:
-        logger.warning(f"An unknown exception ocurred attempting to end {result.args[-1]}")
-        logger.debug(result)
+        logger.warning(f"An  error was received while trying to ending {process_name}")
+        logger.debug(result.stderr)
 
 
-def kill_process(process_executable: str) -> str:
+def kill_process(process_executable: str, force: bool = True) -> subprocess.CompletedProcess[str]:
     """Kill the process, and parse whether it was successfully killed
 
     Args:
         process_executable (str): The process executable to kill
+        force (bool): Whether to force the process termination
 
     Returns:
         str: _description_
     """
     return subprocess.run(
-        ["taskkill", "/f", "/im", process_executable],
+        f"taskkill{' /f' if force else ''} /im \"{process_executable}\"",

```

Comment:
Is there a reason we'd prefer single-string syntax instead of array-based syntax since the OS typically prefers array-based? 🤔

Reply:
Using a string makes it platform-dependent, but we only support Windows anyway. It's faster to insert `"/f"` conditionally IMO, but I can also revert to using an array and conditionally insert it at index 1.

Reply:
I don't believe that's true for most cases with Linux, either, however I would agree it makes it a bit cleaner for the conditional and if we've got no issues with it executing reliable for now, then that works 👌